pkg/controller/datavolume/import-controller.go

```diff
@@ -198,58 +204,56 @@ func (r ImportReconciler) updateAnnotations(dataVolume *cdiv1.DataVolume, annota
 	return errors.Errorf("no source set for import datavolume")
 }
 
-func (r ImportReconciler) updateStatus(log logr.Logger, syncRes dataVolumeSyncResult, syncErr error) (reconcile.Result, error) {
-	if syncErr != nil {
-		return reconcile.Result{}, syncErr
-	}
-	if syncRes.res != nil {
-		return *syncRes.res, nil
+// Reconcile loop for the import data volumes
+func (r ImportReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithValues("DataVolume", req.NamespacedName)
+	return r.updateStatus(r.sync(log, req))
+}
+
+func (r ImportReconciler) sync(log logr.Logger, req reconcile.Request) (dataVolumeSyncResult, error) {
+	var syncRes dataVolumeSyncResult
+	syncErr := r.syncCommon(log, req, &syncRes, nil, nil)
+	if syncErr != nil || syncRes.result != nil {
+		return syncRes, syncErr
 	}
-	//FIXME: pass syncRes instead of args
-	if err := r.reconcileDataVolumePVC(log, syncRes.dv, &syncRes.pvc, syncRes.pvcSpec); err != nil {
-		return reconcile.Result{}, err
+	if syncRes.pvc == nil {
+		if _, dvPrePopulated := syncRes.dv.Annotations[cc.AnnPrePopulated]; !dvPrePopulated {
+			syncRes.pvc, syncErr = r.createPvcForDatavolume(syncRes.dv, syncRes.pvcSpec, r.updateAnnotations)
+		}
 	}
-	return r.reconcileDataVolumeStatus(syncRes.dv, syncRes.pvc, nil, r.updateStatusPhase)
+	if syncRes.pvc != nil && syncErr == nil {
+		syncErr = r.maybeSetPvcMultiStageAnnotation(syncRes.pvc, syncRes.dv)
 
+	}
+	return syncRes, syncErr
 }
 
-func (r *ImportReconciler) reconcileDataVolumePVC(log logr.Logger, dv *cdiv1.DataVolume, pvc **corev1.PersistentVolumeClaim, pvcSpec *v1.PersistentVolumeClaimSpec) error {
-	if _, dvPrePopulated := dv.Annotations[cc.AnnPrePopulated]; dvPrePopulated {
-		return nil
-	}
-	if *pvc == nil {
-		newPvc, err := r.createPvcForDatavolume(dv, pvcSpec, r.updateCheckpoint)
-		if err != nil {
-			if cc.ErrQuotaExceeded(err) {
-				r.updateDataVolumeStatusPhaseWithEvent(cdiv1.Pending, dv, nil, nil,
-					Event{
-						eventType: corev1.EventTypeWarning,
-						reason:    cc.ErrExceededQuota,
-						message:   err.Error(),
-					})
-			}
-			return err
+func (r ImportReconciler) updateStatus(syncRes dataVolumeSyncResult, syncErr error) (reconcile.Result, error) {
+	if syncErr != nil {
+		if cc.ErrQuotaExceeded(syncErr) {
+			r.updateDataVolumeStatusPhaseWithEvent(cdiv1.Pending, syncRes.dv, nil, nil,
+				Event{
+					eventType: corev1.EventTypeWarning,
+					reason:    cc.ErrExceededQuota,
+					message:   syncErr.Error(),
+				})
 		}
-		*pvc = newPvc
-		return nil
+		return reconcile.Result{}, syncErr
```

Comment:
why return without attempting to check/update status?

Reply:
fixed